frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -161,3 +189,264 @@ export const PoolResourceComponent: React.FC<ProvisionerProps> = ({ onParamChang
   }
   return <>{(!loaded || !poolDataLoaded) && <LoadingInline />}</>;
 };
+
+export const StorageClassEncryption: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const [state, dispatch] = React.useReducer(scReducer, scInitialState);
+  const [checked, isChecked] = React.useState(false);
+  const [editKMS, setEditKMS] = React.useState(false);
+  const [validSave, setValidSave] = React.useState(true);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [progress, setInProgress] = React.useState(false);
+  const [currentKMS, setCurrentKMS] = React.useState<KMSConfigMap>(null);
+
+  const csiCMWatchResource: WatchK8sResource = {
+    kind: ConfigMapModel.kind,
+    namespaced: true,
+    isList: false,
+    namespace: CEPH_STORAGE_NAMESPACE,
+    name: KMSConfigMapCSIName,
+  };
+
+  const [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError] = useK8sWatchResource<
+    ConfigMapKind
+  >(csiCMWatchResource);
+
+  React.useEffect(() => {
+    if (!_.isEmpty(csiConfigMap)) {
+      const serviceNames = Object.keys(csiConfigMap?.data);
+      const kmsData = JSON.parse(csiConfigMap?.data[serviceNames[serviceNames.length - 1]]);
+      setCurrentKMS(kmsData);
+      const url = parseURL(kmsData.VAULT_ADDR);
+      const kmsObj: KMSConfig = {
+        name: {
+          value: kmsData.KMS_SERVICE_NAME,
+          valid: true,
+        },
+        address: {
+          value: `${url.protocol}//${url.hostname}`,
+          valid: true,
+        },
+        port: {
+          value: url.port,
+          valid: true,
+        },
+        backend: kmsData.VAULT_BACKEND_PATH,
+        caCert: state.kms.caCert ?? null,
+        caCertFile: state.kms.caCertFile,
+        tls: kmsData.VAULT_TLS_SERVER_NAME,
+        clientCert: state.kms.clientCert ?? null,
+        clientCertFile: state.kms.clientCertFile,
+        clientKey: state.kms.caCert ?? null,
+        clientKeyFile: state.kms.clientKeyFile,
+        providerNamespace: kmsData.VAULT_NAMESPACE,
+        hasHandled: true,
+      };
+      setEncryptionDispatch(SCActionType.SET_KMS_ENCRYPTION, '', dispatch, kmsObj);
```

Comment:
you should be dispatching on click of an action ideally .
This will set the state whenever map is not empty -> which does not seems like ideal condition

Reply:
The reason for doing this is to set initial state of component with respect to value of configMap, so as config map loads, we have to set the state.